packages/headless/contributors/adding-a-sub-package.md

```diff
@@ -0,0 +1,36 @@
+# Adding a sub-package
+
+Headless provides exports through multiple sub packages. A sub-package groups together exports (i.e. controllers, actions, reducers, engines) that work together as a cohesive unit. By separating exports into sub-packages, it becomes clear to users of headless what exports are available to build a use-case.
+
+
+## To add a new sub-package:
+
+1. Create an entry file for your sub-package inside the `src` directory (e.g. case-assist.ts).
+2. Configure nodejs and browser bundles inside `rollup.config.js` for your entry file.
+3. Inside the `pkg` directory, create a new directory having the name of your sub-package.
+4. Inside the new directory, add a `package.json` file and fill in the paths to your bundled files and type definitions.
+
+```json
+pkg/case-assist/package.json
+
+{
+  "name": "case-assist",
+  "version": "1.0.0",
+  "description": "Headless Case Assist Module",
+  "main": "../../dist/case-assist/headless.js",
+  "module": "../../dist/case-assist/headless.esm.js",
+  "browser": "../../dist/browser/case-assist/headless.esm.js",
+  "types": "../../dist/definitions/case-assist.d.ts",
+  "license": "Apache-2.0"
+}
+```
+
+## Testing your sub-package:
+
+1. Build the headless project: `npm run build`.
+2. Create a tarball: `npm pack`.
+3. Install the tarball as a dependency of a different project: `npm i <path to the tarball>`.
```

Comment:
The commands are similar, but I think `npm pack` is more accurate because it creates an output matching what is uploaded npm.

For example, this config in the [package.json](https://github.com/coveo/ui-kit/pull/717/files#diff-b49fb4b1d19b4b6568e5ff36ec318c7410ab21ce4405d21930510332d64c3905R15) tells npm what files to include in the tarball. Say someone created a new directory, but did not update the config. I suspect a local symlink would provide access to the new directory, but when installing from npm, the directory would be missing.

Reply:
Ok. 

I just know that personally, I would hate to work with that flow and would not do as described here :D 

Seems like a hassle for "maybe it's going to be more accurate".